### PR TITLE
Support websocket

### PR DESCRIPTION
**Pull request by kkoomen**
This PR adds support for the websocket protocols. The functionality includes support for websockets and socket.io.

### CODE REVIEW

File: src/throttler.guard.ts [hunk 1]
```diff
@@ -13,16 +13,24 @@ import { ThrottlerOptions } from './throttler.interface';
...
+  storageService: ThrottlerStorage;
+
   constructor(
     @Inject(THROTTLER_OPTIONS) private readonly options: ThrottlerOptions,
-    @Inject(ThrottlerStorage) private readonly storageService: ThrottlerStorage,
+    @Inject(ThrottlerStorage) storageService: ThrottlerStorage,
     private readonly reflector: Reflector,
-  ) {}
+  ) {
+    this.storageService = options.storage || storageService;
```
Comment (jmcdo29) [inline src/throttler.guard.ts:23]: This shouldn't be necessary due to how the `ThrottlerCoreModule` works with setting the `ThrottlerStorage` provider

File: src/throttle.decorator.ts [hunk 2]
```diff
@@ -1,25 +1,33 @@
 import { THROTTLER_LIMIT, THROTTLER_SKIP, THROTTLER_TTL } from './throttler.constants';
 
-function setThrottlerMetadata(target: Function, limit: number, ttl: number): void {
```
Comment (jmcdo29) [inline src/throttle.decorator.ts:3]: Any reason to remove this function? Figured having it as a separate function would make it easier to modify if need be

File: test/app/gateways/app.gateway.ts [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+import { SkipThrottle, Throttle, ThrottlerGuard } from '../../../src';
 import { AppService } from '../app.service';
-import { SkipThrottle, Throttle } from '../../../src';
 
 @Throttle(2, 10)
 @WebSocketGateway({ path: '/' })
 export class AppGateway {
   constructor(private readonly appService: AppService) {}
+
+  @UseGuards(ThrottlerGuard)
```
Comment (jmcdo29) [inline test/app/gateways/app.gateway.ts:11]: We can set this on the class rather than at the method level

File: src/throttler.guard.ts (hunk 1 above)
Comment (kkoomen) [reply src/throttler.guard.ts:23]: Mhm, couldn't get it to work earlier though.

File: src/throttle.decorator.ts (hunk 2 above)
Comment (kkoomen) [reply src/throttle.decorator.ts]: reverted it.

**Comment by jmcdo29** [comment]
I merged this manually from the command line. Closing the PR.

**Pull request closed without merging**